google-beta/resource_dataflow_flex_template_job.go

```diff
@@ -177,10 +174,96 @@ func resourceDataflowFlexTemplateJobRead(d *schema.ResourceData, meta interface{
 	return nil
 }
 
-// resourceDataflowFlexTemplateJobUpdate is a blank method to enable updating
-// the on_delete virtual field
+func waitForDataflowJobState(d *schema.ResourceData, config *Config, jobID, userAgent string, timeout time.Duration, targetState string) error {
+	return resource.Retry(timeout, func() *resource.RetryError {
+		project, err := getProject(d, config)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		region, err := getRegion(d, config)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		job, err := resourceDataflowJobGetJob(config, project, region, userAgent, jobID)
+		if err != nil {
+			if isRetryableError(err) {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+
+		state := job.CurrentState
+		if state == targetState {
+			log.Printf("[DEBUG] the job with ID %q has state %q.", jobID, state)
+			return nil
+		}
+		if _, terminated := dataflowTerminalStatesMap[state]; terminated {
+			return resource.NonRetryableError(fmt.Errorf("the job with ID %q has terminated with state %q instead of expected state %q", jobID, state, targetState))
+		} else {
+			log.Printf("[DEBUG] the job with ID %q has state %q.", jobID, state)
+			return resource.RetryableError(fmt.Errorf("the job with ID %q has state %q, waiting for %q", jobID, state, targetState))
+		}
+	})
+}
+
+// resourceDataflowFlexTemplateJobUpdate updates a Flex Template Job resource.
 func resourceDataflowFlexTemplateJobUpdate(d *schema.ResourceData, meta interface{}) error {
-	return nil
+	// Don't send an update request if only virtual fields have changes
+	if resourceDataflowJobIsVirtualUpdate(d, resourceDataflowFlexTemplateJob().Schema) {
```

Comment:
I don't think this resource has virtual fields, so this can probably be removed

Reply:
Just like with dataflow_job, dataflow_flex_template_job has a virtual "on_delete" field. It is not used when creating or updating a job. It is only used when the job resource should be deleted, and governs whether to drain or cancel the job (the two ways of stopping a running job).